### PR TITLE
feat(invite-guest): invite guest to organization

### PR DESCRIPTION
**Pull request by enigbe**
### What does this PR do?
- Check the status of a guest's email
- Creates and verifies a new user from UUID sent to guest's email and guest-inputted password
- Adds guest to organization
### Task completed
- Check guest's status: `GET '/organizations/{id}/invites/{uuid}'`
- Create user from UUID: `POST '/guests/invite'`
- Add guest (now a new user) to organization: `POST '/organizations/{id}/guests/{uuid}'`

### How should this be tested?
With Postman/Insomnia
- `GET '/organizations/{id}/invites/{uuid}'` where `id` is the organization's ID and `uuid` is the UUID sent to the guest's email from `POST '/organizations/{id}/send-invite'`
- `POST '/users/invite'` with request body `{"uuid": <UUID>, "password": "password"}` where `UUID` is of the form `496401b2-eaa5-4d8f-9dee-38d8441b524c`
- `POST '/organizations/{id}/guests/{uuid}'`

### CODE REVIEW

File: user/user.go [hunk 1]
```diff
@@ -275,3 +275,87 @@ func (uh *UserHandler) GetUserOrganizations(response http.ResponseWriter, reques
...
+	}
+
+	user := &User{
+		Email:             email,
+		Password:          hashPassword,
+		IsVerified:        true,
+		EmailVerification: con,
+		CreatedAt:         time.Now(),
+		Deactivated:       false,
+		Social:            false,
```
Comment (Aminujb) [inline user/user.go:348]: Is social a boolean field on user account? I am of the opinion that we take it down as it does nothing on a user instance. It is only used on the member instance.

File: organizations/user.go [hunk 2]
```diff
@@ -614,3 +614,46 @@ func (oh *OrganizationHandler) ReactivateMember(w http.ResponseWriter, r *http.R
...
+		return
+	}
+
+	// 1. Query organization invites collection for uuid
+	res, err := utils.GetMongoDbDoc(OrganizationInviteCollection, bson.M{"uuid": guestUUID})
+	if err != nil {
+		utils.GetError(err, http.StatusBadRequest, w)
+		return
+	}
+	fmt.Println(res)
```
Comment (Aminujb) [inline organizations/user.go:634]: Please take out print statements.

File: user/user.go [hunk 3]
```diff
@@ -275,3 +275,87 @@ func (uh *UserHandler) GetUserOrganizations(response http.ResponseWriter, reques
 
 	utils.GetSuccess("user organizations retrieved successfully", orgs, response)
 }
+
+// Create a new user from UUID guest invite sent to user and a supplied password.
+func (uh *UserHandler) CreateUserFromUUID(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("content-type", "application/json")
+
+	user_collection, orgInvite := "users", "organizations_invites"
```
Comment (Aminujb) [inline user/user.go:283]: Lets also move these constants to the model file. You can do this correction other relevant places in the user file. I started this in organization too.

File: user/user.go [hunk 4]
```diff
@@ -275,3 +275,87 @@ func (uh *UserHandler) GetUserOrganizations(response http.ResponseWriter, reques
...
+
+	type UUIDPassword struct {
```
Comment (Aminujb) [inline user/user.go:285]: Can this live in the model file too?

File: main.go [hunk 5]
```diff
@@ -131,6 +134,7 @@ func Router(Server *socketio.Server) *mux.Router {
 	r.HandleFunc("/users/{user_id}", auth.IsAuthenticated(user.DeleteUser)).Methods("DELETE")
 	r.HandleFunc("/users", auth.IsAuthenticated(user.GetUsers)).Methods("GET")
 	r.HandleFunc("/users/{email}/organizations", auth.IsAuthenticated(user.GetUserOrganizations)).Methods("GET")
+	r.HandleFunc("/users/invite", user.CreateUserFromUUID).Methods("POST")
```
Comment (Aminujb) [inline main.go:137]: This invite, are we inviting people to join zuri chat and also handling inviting people to join an org on zuri chat?

**Review by Aminujb — CHANGES_REQUESTED** [review]

File: user/user.go (hunk 1 above)
Comment (enigbe) [reply user/user.go:348]: Okay. I will modify the field.

File: organizations/user.go (hunk 2 above)
Comment (enigbe) [reply organizations/user.go:636]: Noted.

File: user/user.go (hunk 3 above)
Comment (enigbe) [reply user/user.go:283]: I will make the necessary changes

File: main.go (hunk 5 above)
Comment (enigbe) [reply main.go:140]: This is built in response to the invite that guests receive to join an organization on zurichat. It serves a second purpose of adding and verifying new users to zurichat.

Comment (Aminujb) [reply main.go:140]: Okay, great.

**Pull request merged**